### PR TITLE
BaSP-TG-7: Employees list and delete

### PR DESCRIPTION
**Pull request by Martinzancocchia**
Employees List, Delete and modals added

### CODE REVIEW

File: src/Components/Employees/index.js [hunk 1]
```diff
@@ -1,27 +1,84 @@
...
+      <Modal
+        show={showModal}
+        handleModal={saveShowModal}
+        deleteEmployee={deleteEmployee}
+        employee={selectedEmployee}
+      />
+      <a href="/employees/form" className={styles.button}>
+        Add new employee +
+      </a>
+      <table className={styles.table}>
```
Comment (martumoser) [inline src/Components/Employees/index.js:50]: The table and the rows should be another react component as was showing on the class.

File: src/Components/Employees/index.js [hunk 2]
```diff
@@ -1,27 +1,84 @@
...
   }, []);
 
+  // const updateEmployee = () => {
+  //   fetch(`${process.env.REACT_APP_API_URL}/employees`)
+  //     .then((response) => response.json())
+  //     .then((response) => {
+  //       saveEmployees(response.data);
+  //     });
+  // };
+
```
Comment (martumoser) [inline src/Components/Employees/index.js:25]: Please don't leave comments on the PR. I think that you are using this lines of code on the other pr but they are not necessary on this pr

File: src/Components/Employees/index.js [hunk 3]
```diff
@@ -1,27 +1,84 @@
...
+  const deleteEmployee = async (id) => {
+    fetch(`${process.env.REACT_APP_API_URL}/employees/${id}`, {
+      method: 'DELETE'
+    });
+    saveSelection(employees.filter((employee) => employee._id !== id));
+  };
+
+  // eslint-disable-next-line no-unused-vars
```
Comment (martumoser) [inline src/Components/Employees/index.js:33]: please delete this comment

File: src/Components/Employees/Modal/modal.module.css [hunk 4]
```diff
@@ -0,0 +1,65 @@
+.container {
+  position: absolute;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  width: 100vw;
+  height: 100vh;
+  background-color: rgba(0, 0, 0, 0.5);
```
Comment (martumoser) [inline src/Components/Employees/Modal/modal.module.css:8]: please replace this color to color hex to keep consistency with the other colors

File: src/Components/Employees/Modal/index.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    deleteEmployee(employee.id);
+    closeModal();
+    window.location.reload();
+    alert('Employee deleted');
+  };
+
+  if (!show) {
+    return null;
+  }
+  return (
```
Comment (martumoser) [inline src/Components/Employees/Modal/index.js:16]: please leave a space between lines 15 and 16

File: src/Components/Employees/Modal/index.js [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    <div className={styles.container}>
+      <div className={styles.modal}>
+        <button onClick={() => closeModal()} className={styles.close}>
+          &times;
+        </button>
+        <h3>Message from Trackgenix</h3>
+        <small>Once deleted, the changes cannot be undone.</small>
```
Comment (martumoser) [inline src/Components/Employees/Modal/index.js:23]: avoid use this html element. You can use css to add styles to the elements

**Review by martumoser — CHANGES_REQUESTED** [review]
Please add a description to the PR

File: src/Components/Employees/index.js (hunk 2 above)
Comment (Martinzancocchia) [reply src/Components/Employees/index.js:25]: done

File: src/Components/Employees/index.js (hunk 3 above)
Comment (Martinzancocchia) [reply src/Components/Employees/index.js:33]: done

File: src/Components/Employees/Modal/modal.module.css (hunk 4 above)
Comment (Martinzancocchia) [reply src/Components/Employees/Modal/modal.module.css:8]: done

File: src/Components/Employees/Modal/index.js (hunk 5 above)
Comment (Martinzancocchia) [reply src/Components/Employees/Modal/index.js:16]: done

File: src/Components/Employees/Modal/index.js (hunk 6 above)
Comment (Martinzancocchia) [reply src/Components/Employees/Modal/index.js:23]: done

File: src/Components/Employees/List/index.js [hunk 7]
```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import styles from '../employees.module.css';
+
+const employeeList = (props) => {
```
Comment (martumoser) [inline src/Components/Employees/List/index.js:4]: Remember that react components starts with uppercase and also the list has some employee so it should be plural: EmployeesList

Comment (martumoser) [inline src/Components/Employees/List/index.js:4]: Please destructure the props to know which are the props. Do the same that you did on Modal component

File: src/Components/Employees/Modal/index.js [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+
+  if (!show) {
+    return null;
+  }
+
+  return (
+    <div className={styles.container}>
+      <div className={styles.modal}>
+        <button onClick={() => closeModal()} className={styles.close}>
+          &times;
```
Comment (martumoser) [inline src/Components/Employees/Modal/index.js:21]: This is not necessary

**Review by martumoser — CHANGES_REQUESTED** [review]

File: src/Components/Employees/List/index.js (hunk 7 above)
Comment (Martinzancocchia) [reply src/Components/Employees/List/index.js:4]: Done

**Review by martumoser — APPROVED** [review]

**Pull request merged**